src/waveform/renderers/allshader/digitsrenderer.cpp

```diff
@@ -129,13 +127,28 @@ void allshader::DigitsRenderer::updateTexture(
         }
     } while (retry);
 
-    m_height = static_cast<float>(std::ceil(maxTextHeight) + space * 2 + 1);
+    m_height = static_cast<float>(std::ceil(maxTextHeight)) + space * 2.f + 1.f;
 
-    // Space around the digits
-    totalTextWidth += (space * 2 + 1) * NUM_CHARS;
-    totalTextWidth = std::ceil(totalTextWidth);
+    const float y = maxTextHeight + space - 0.5f;
 
-    const qreal y = maxTextHeight + space - 0.5;
+    auto roundToPixel = makeRoundToPixel(devicePixelRatio);
```

Comment:
```suggestion
    auto roundToPixelFnc = makeRoundToPixelFnc(devicePixelRatio);
```

Reply:
I will rename makeRoundToPixel to something more explicit, but I don't see a reason to rename roundToPixel, as it is clearly a function and called as such.

I hope that with that did this can be merged, because I need to rebase my rendergraph continuation on this.